pyjob/sge.py

```diff
@@ -65,9 +73,61 @@ def info(self):
                     data[kv[0]] = kv[1]
         return data
 
-    def close(self):
-        """Close this :obj:`~pyjob.sge.SunGridEngineTask` after completion"""
-        self.wait()
+    @classmethod
+    def get_sge_avail_configs(cls, param):
+        """Get the set of available configurations for a given SGE parameter
+
+        Parameters
+        ----------
+        param : :obj:~SGEConfigParameter
+            The parameter to be tested
+
+        Returns
+        -------
+        set
+            A set with the available configurations for the parameter of interest
+
+        Raises
+        ------
+        :exc:`ValueError`
+           Parameter is not found in :obj:~SGEConfigParameter
+
+        """
+
+        if param in cls._sge_avail_configs_by_env:
+            return cls._sge_avail_configs_by_env[param]
+
+        if SGEConfigParameter(param) == SGEConfigParameter.ENVIRONMENT:
+            cmd = ['qconf', '-spl']
+        elif SGEConfigParameter(param) == SGEConfigParameter.QUEUE:
+            cmd = ["qconf", '-sql']
+        else:
+            raise ValueError('Requested SGE parameter is not supported!')
+
+        stdout = cexec(cmd, permit_nonzero=True)
+        config = []
+        for line in stdout.splitlines():
+            line = line.split()
+            if len(line) > 1:
+                return config
+            else:
+                config.append(line[0].encode('utf-8'))
+
+        cls._sge_avail_configs_by_env[param] = set(config)
+        return cls._sge_avail_configs_by_env[param]
+
+    def _check_requirements(self):
+        """Check if the requirements for task execution are met"""
+
+        self._ensure_exec_available('qstat')
+
+        if self.environment and self.environment not in self.get_sge_avail_configs(SGEConfigParameter.ENVIRONMENT):
+            raise PyJobError('Requested environment {} cannot be found. List of available environments: {}'
+                             ''.format(self.environment, self.get_sge_avail_configs(SGEConfigParameter.ENVIRONMENT)))
+
+        if self.queue and self.queue not in self.get_sge_avail_configs(SGEConfigParameter.QUEUE):
+            raise PyJobError('Requested queue {} cannot be found. List of available queues: {}'
```

Comment:
I know the result of the function is retrieved from cache but I would still store it and then use it afterwards.

```python
sge_config_by_env = self.get_sge_avail_configs(SGEConfigParameter.ENVIRONMENT)
if self.environment and self.environment not in sge_config_by_env:
    raise PyJobError('Requested environment {} cannot be found. List of available environments: {}'.format(self.environment, sge_config_by_env.keys()))
```